### ISSUE TITLE
Странныей символы в консоли при работе.

### ISSUE THREAD

**Issue by kt368** [porthole-ascend-cinnamon/mhddos_proxy#72]
Win10 x64, не docker (нативно через python runner.py).
Запустил атаку, в консоли вижу странные символы 93m, 94, 39m.
```
[13:48:06 - INFO] ←[93mЦіль:←[94m www.cdek.ru,←[93m Порт:←[94m 443,←[93m Метод:←[94m GET←[93m Потоків:←[94m 250←[93m PPS:←[94m 0,←[93m BPS:←[94m 0←[39m
[13:48:06 - INFO] ←[93mЦіль:←[94m www.cdek.ru,←[93m Порт:←[94m 443,←[93m Метод:←[94m STRESS←[93m Потоків:←[94m 250←[93m PPS:←[94m 57,←[93m BPS:←[94m 601.17 kBit←[39m
[13:48:06 - INFO] ←[93mЦіль:←[94m cdek.by,←[93m Порт:←[94m 443,←[93m Метод:←[94m GET←[93m Потоків:←[94m 250←[93m PPS:←[94m 0,←[93m BPS:←[94m 0←[39m
[13:48:06 - INFO] ←[93mЦіль:←[94m cdek.by,←[93m Порт:←[94m 443,←[93m Метод:←[94m STRESS←[93m Потоків:←[94m 250←[93m PPS:←[94m 0,←[93m BPS:←[94m 0←[39m
[13:48:06 - INFO] ←[93mНовий цикл через: ←[94m448 секунд←[39m
[13:48:06 - INFO] ←[93mКількість проксі: ←[94m295←[39m
[13:48:08 - INFO] ←[93mЦіль:←[94m www.cdek.ru,←[93m Порт:←[94m 443,←[93m Метод:←[94m GET←[93m Потоків:←[94m 250←[93m PPS:←[94m 301,←[93m BPS:←[94m 1.48 MBit←[39m
[13:48:08 - INFO] ←[93mЦіль:←[94m www.cdek.ru,←[93m Порт:←[94m 443,←[93m Метод:←[94m STRESS←[93m Потоків:←[94m 250←[93m PPS:←[94m 191,←[93m BPS:←[94m 1.96 MBit←[39m
[13:48:08 - INFO] ←[93mЦіль:←[94m cdek.by,←[93m Порт:←[94m 443,←[93m Метод:←[94m GET←[93m Потоків:←[94m 250←[93m PPS:←[94m 0,←[93m BPS:←[94m 0←[39m
[13:48:08 - INFO] ←[93mЦіль:←[94m cdek.by,←[93m Порт:←[94m 443,←[93m Метод:←[94m STRESS←[93m Потоків:←[94m 250←[93m PPS:←[94m 66,←[93m BPS:←[94m 663.95 kBit←[39m
[13:48:08 - INFO] ←[93mНовий цикл через: ←[94m446 секунд←[39m
[13:48:08 - INFO] ←[93mКількість проксі: ←[94m295←[39m
```
Вот скриншот:
![image](https://user-images.githubusercontent.com/7873883/161846282-7f5854a5-bfa0-4f8d-a788-769dd3799b34.png)
Пробовал запускать через стандартный cmd.exe и через PowerShell, поведение одинаковое.

**Comment by porthole-ascend-cinnamon** [comment]
Остання версія? pip install зробили? 
Це не підтягнулися кольори, але дивно чому...
Виконайте будь ласка цей код і надішліть результат.
`python -c 'import os; print(os.name)'`

**Comment by MayborodaYuriy** [comment]
Мені у свій час допомогло створення ключа `VirtualTerminalLevel` в реєстрі `HKEY_CURRENT_USER/Console` типу `DWORD` зі значенням `1`.

**Comment by kt368** [comment]
> 

Да, версия последняя.
`python -m pip install -r requirements.txt` делал
`python -c 'import os; print(os.name)'` выдаёт ошибку:
```
  File "<string>", line 1
    'import
    ^
SyntaxError: unterminated string literal (detected at line 1)
```
Тогда отдельно запустил python, в нём выполнил
```
>>> import os;
>>> print(os.name);
nt
```

**Comment by kt368** [comment]
> Мені у свій час допомогло створення ключа `VirtualTerminalLevel` в реєстрі `HKEY_CURRENT_USER/Console` типу `DWORD` зі значенням `1`.

не помоголо(

**Comment by porthole-ascend-cinnamon** [comment]
Давайте ще таке спробуємо. В python:
```
from colorama import Fore, init
init()
print(Fore.RED + 'some red text')
print(Fore.LIGHTRED_EX + 'some light red text')
```

**Comment by kt368** [comment]
Цвета начали отображаться после установки
`pip install termcolor`
Для того чтоб понять где собака зарыта удалил `termcolor` и выполнил ваши команды. Текст вывелся красным и светло красным текстом, всё как положено.
![image](https://user-images.githubusercontent.com/7873883/161859786-9dd70af4-5f95-4a8d-937d-efef296b3bd4.png)
Заново установил `termcolor`, запустил ддос атаку и снова всё заработало как и положено, в цвете.

**Comment by porthole-ascend-cinnamon** [comment]
Так, не понял, в итоге и без установленного `termcolor` работали цвета в консоли, но при это не работали при атаке?

**Comment by kt368** [comment]
Да, именно так. Цвет текста при выполнениия ваших команд тремя комментами выше был правильный и без `termcolor`, а при атаке цвета не отображались пока не установил этот самый `termcolor`.

**Comment by porthole-ascend-cinnamon** [comment]
Очень странно конечно...Попробуйте последнюю версию пожалуйста

**Comment by alexnest-ua** [comment]
> Win10 x64, не docker (нативно через python runner.py). Запустил атаку, в консоли вижу странные символы 93m, 94, 39m.
> 
> ```
> [13:48:06 - INFO] ←[93mЦіль:←[94m www.cdek.ru,←[93m Порт:←[94m 443,←[93m Метод:←[94m GET←[93m Потоків:←[94m 250←[93m PPS:←[94m 0,←[93m BPS:←[94m 0←[39m
> [13:48:06 - INFO] ←[93mЦіль:←[94m www.cdek.ru,←[93m Порт:←[94m 443,←[93m Метод:←[94m STRESS←[93m Потоків:←[94m 250←[93m PPS:←[94m 57,←[93m BPS:←[94m 601.17 kBit←[39m
> [13:48:06 - INFO] ←[93mЦіль:←[94m cdek.by,←[93m Порт:←[94m 443,←[93m Метод:←[94m GET←[93m Потоків:←[94m 250←[93m PPS:←[94m 0,←[93m BPS:←[94m 0←[39m
> [13:48:06 - INFO] ←[93mЦіль:←[94m cdek.by,←[93m Порт:←[94m 443,←[93m Метод:←[94m STRESS←[93m Потоків:←[94m 250←[93m PPS:←[94m 0,←[93m BPS:←[94m 0←[39m
> [13:48:06 - INFO] ←[93mНовий цикл через: ←[94m448 секунд←[39m
> [13:48:06 - INFO] ←[93mКількість проксі: ←[94m295←[39m
> [13:48:08 - INFO] ←[93mЦіль:←[94m www.cdek.ru,←[93m Порт:←[94m 443,←[93m Метод:←[94m GET←[93m Потоків:←[94m 250←[93m PPS:←[94m 301,←[93m BPS:←[94m 1.48 MBit←[39m
> [13:48:08 - INFO] ←[93mЦіль:←[94m www.cdek.ru,←[93m Порт:←[94m 443,←[93m Метод:←[94m STRESS←[93m Потоків:←[94m 250←[93m PPS:←[94m 191,←[93m BPS:←[94m 1.96 MBit←[39m
> [13:48:08 - INFO] ←[93mЦіль:←[94m cdek.by,←[93m Порт:←[94m 443,←[93m Метод:←[94m GET←[93m Потоків:←[94m 250←[93m PPS:←[94m 0,←[93m BPS:←[94m 0←[39m
> [13:48:08 - INFO] ←[93mЦіль:←[94m cdek.by,←[93m Порт:←[94m 443,←[93m Метод:←[94m STRESS←[93m Потоків:←[94m 250←[93m PPS:←[94m 66,←[93m BPS:←[94m 663.95 kBit←[39m
> [13:48:08 - INFO] ←[93mНовий цикл через: ←[94m446 секунд←[39m
> [13:48:08 - INFO] ←[93mКількість проксі: ←[94m295←[39m
> ```
> 
> Вот скриншот: ![image](https://user-images.githubusercontent.com/7873883/161846282-7f5854a5-bfa0-4f8d-a788-769dd3799b34.png) Пробовал запускать через стандартный cmd.exe и через PowerShell, поведение одинаковое.

Але перед тим налаштувати консоль, щоб вона виводила кольоровий текст та без каракуль (які все-одно ніяк не впливають на ефективність програми):
1) Відкрити нову cmd (Командая строка) Від імені адміністратора

2) Вставити наступне та натиснути Enter:

reg add HKEY_CURRENT_USER\Console /v VirtualTerminalLevel /t REG_DWORD /d 0x00000001 /f
Якщо не спрацювало(після пунку 5 у вас не кольоровий вивід), спробуйте цю команду:

reg add HKEY_CURRENT_USER\Console /v VirtualTerminalLevel /t REG_DWORD /d 0x00000000 /f


Також рекомендую встановити останню версію на ПК, починаючи з пункту 4 (який перенаправить вас на пункт 3) звідси:
https://telegra.ph/Vstanovlennya-mhddos-proxy-napryamu-na-vash-komp-03-27
Інфа, що вище теж звідти взята

**Comment by porthole-ascend-cinnamon** [comment]
@alexnest-ua але ж має працювати і без того по ідеї...colorama призначена власе щоби каракулі транслювати в системні виклики Windows...

**Comment by alexnest-ua** [comment]
> Заработало. Удалил `termcolor`. Скачал последнюю версию `mhddos_proxy`, теперь цвета отображаются правильно. Спасибо большое! З.Ы. Команду `reg add HKEY_CURRENT_USER\Console /v VirtualTerminalLevel /t REG_DWORD /d 0x00000001 /f` на всякий случай выполнил. ![image](https://user-images.githubusercontent.com/7873883/162035399-ffc0d6a5-4481-4974-9525-7b6e1cf666ed.png)

було б не погано, якби ви спочатку все оновили - спробували по дефолту, а потім інші команди робили, але ок...

Думаю таки проблема у версії була
